tests/test_vtk.py

```diff
@@ -35,6 +36,22 @@ def test_validation_pluto(fn):
     assert PlutoVtkDataset._is_valid(fn)
 
 
+@pytest.mark.parametrize(("fn", "geometry"), PLUTO_VTK_FILES)
+def test_parse_pluto_metadata(fn, geometry):
+    ds = yt_idefix.load(fn)
+    assert ds._definitions_header == os.path.join(ds.fullpath, "definitions.h")
+    assert "unit_length" in ds.parameters
+    assert "unit_density" in ds.parameters
+    assert "unit_velocity" in ds.parameters
+    assert ds.current_time != -1
+    assert ds.geometry == "cartesian"
+    assert ds.length_unit / ds.velocity_unit == ds.time_unit
+    with pytest.raises(ValueError):
+        ds = yt_idefix.load(fn, definitions_header="definitions2.h")
+    with pytest.warns(UserWarning):
+        ds = yt_idefix.load(fn, definitions_header="definitions2.h", geometry=geometry)
```

Comment:
a couple remarks here
1) these looks like 2 different tests, it'd be preferable to separate them from the main one, for a total of 3 test functions
2) please use the `match` argument in `pytest.warns` and `pytest.raises` to match the string you're expecting. If it's an exact string, it should be straighforward, but beware that `match=` expects a regexp pattern in general. Here again, I'm happy to do it for you if you can at least tell me what the expected errors and warnings look like.

Reply:
Here I want to test the occasions when some arguments are missed or assigned incorrectly, which are expected in my `if-elif-else` statements in `_parse_header_file()`. The exact strings for `ValueError` and `warnings` are written in `_parse_header_file()`. So I think maybe it's verbose to write them here again?

Reply:
I just got your point! I'll improve it to make the test more accurate. Thanks!

> 1. these looks like 2 different tests, it'd be preferable to separate them from the main one, for a total of 3 test functions

Besides, as I said, these two tests are designed for two expected cases during parsing metadata, should I separate them?

Reply:
> So I think maybe it's verbose to write them here again?

It doesn't matter if it's verbose, the role such a test is to ensure we're getting the expected error. Matching the string ensures we're not passing the test by accident because some _different_ `ValueError` is raised. Besides, tests read like a form of in depth documentation, if I can't tell what's the error without looking at the current implementation then it's more of a pain to maintain.
In general, test code doesn't have to be pretty, or minimal, or [DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself); what's important is that it's clear and as maintainable as possible

Reply:
I see. Thanks for your explainations again! I've learned a lot from this work. I wrote these test just after knowing some basic usage and I'll learn more details later!

Reply:
> Besides, as I said, these two tests are designed for two expected cases during parsing metadata, should I separate them?

Yes, it's okay to have several tests related in topics. What's important here is that we avoid shadowing a failure with another in case both are broken at the same time